docs/puma_support.md

```diff
@@ -0,0 +1,66 @@
+# Puma Support Policy
+
+## What
+
+This document's purpose is to help users and contributors of this library make future plans based on the trajectory of this project. It is loosely worded and should be expected to change over time. Maintainers and contributors are encouraged to follow the guidance here but are not strictly bound by it.
+
+## What does "not supported" mean?
+
+Puma is a community-supported and run project. Support entitles the community to open issues and file PRs. It does not guarantee or warranty any action regarding those issues and PRs. All issues and PRs can be closed at a maintainer's discretion, regardless of whether they qualify under "support." If an issue falls outside of Puma support, maintainers are not obliged to keep it open, and if a PR falls outside of Puma support, maintainers are not obliged to review it.
+
+## Puma release support policy
+
+By default, Puma supports one release version, the latest release. That means if there is a bug in 3.2.1 (made-up numbers for example only), but the latest release is 4.3.2, the bug must be reproduced in 4.3.2 to qualify for support.
+
+When a new major (and sometimes minor) version is released, Puma maintainers will try to preserve the status of the old version via branches and/or tags. Puma maintainers can choose to backport fixes to older versions and cut a release of older versions, but this behavior should not be assumed.
+
+If a bug report cannot be reproduced with the latest Puma version, it may be closed.
+
+## Ruby version support policy
+
+ "Ruby version" refers to the https://github.com/ruby/ruby implementation. Puma also supports other Ruby implementations (the origin of Puma involves an alternate Ruby implementation, rubinius) see below.
+
+Puma supports "Ruby upstream + 2 versions". Ruby upstream support is documented at this page https://www.ruby-lang.org/en/downloads/branches/. At this time of writing, it is 3 "minor" versions, where a minor version is X.Y with X.Y.Z being a patch release. If a Ruby version is not EOL, it is supported. Because we support "+ 2 versions", if 3.2 is the last non-EOL version, Puma will support 3.1 and 3.0 as well. The Puma team recommends you always use the latest Ruby release.
+
+When Puma supports a Ruby version, it will include that version in the CI matrix. The gemspec specifies the minimum supported Ruby versions. If a bug is reported, the report must be reproducible on a currently supported Ruby version.
+
```

Comment:
```suggestion
When Puma supports a Ruby version, it will include that version in the CI matrix. The gemspec specifies the minimum supported Ruby versions. If a bug is reported, the report must be reproducible using the most recent patch version of a currently supported Ruby minor version.

```

Reply:
As discussed, maybe some additional text referring to 'most recent Ruby patch version'.

Not sure if the above is clear or whether an example would help.  Something like:

As an example, the current Ruby 3.4 patch version is 3.4.5.  If a Puma bug is found with Ruby 3.4.4 (or earlier) that isn’t reproducible with Ruby 3.4.5, it is up to the Puma team to decide if a Puma fix will be accepted.